tests/nnstreamer_plugins/unittest_plugins.cc

```diff
@@ -3344,8 +3412,274 @@ TEST (test_tensor_filter, reload_tflite_same_model_wrong_dims_n)
   g_free (test_model_renamed);
 }
 
+/**
+ * @brief Test framework auto detecion option in tensor-filter.
+ */
+TEST (test_tensor_filter, framework_auto_ext_tflite_01)
+{
+  GET_MODEL_PATH (mobilenet_v1_1.0_224_quant.tflite)
+
+  str_launch_line = g_strdup_printf ("videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter name=tfilter framework=auto model=%s ! tensor_sink", test_model);
+  TEST_TENSOR_FILTER_AUTO_OPTION_P (str_launch_line, tensorflow-lite)
+}
+
+/**
+ * @brief Test framework auto detecion option in tensor-filter.
+ * @details The order of tensor filter options has changed.
+ */
+TEST (test_tensor_filter, framework_auto_ext_tflite_02)
+{
+  GET_MODEL_PATH (mobilenet_v1_1.0_224_quant.tflite)
+
+  str_launch_line = g_strdup_printf ("videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter name=tfilter model=%s framework=auto ! tensor_sink", test_model);
+  TEST_TENSOR_FILTER_AUTO_OPTION_P (str_launch_line, tensorflow-lite)
+}
+
+/**
+ * @brief Test framework auto detecion option in tensor-filter.
+ * @details Test if options are insensitive to the case
+ */
+TEST (test_tensor_filter, framework_auto_ext_tflite_03)
+{
+  GET_MODEL_PATH (mobilenet_v1_1.0_224_quant.tflite)
+
+  str_launch_line = g_strdup_printf ("videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter name=tfilter model=%s framework=AutO ! tensor_sink", test_model);
+  TEST_TENSOR_FILTER_AUTO_OPTION_P (str_launch_line, tensorflow-lite)
+}
+
+/**
+ * @brief Test framework auto detecion option in tensor-filter.
+ * @details Negative case when model file does not exist
+ */
+TEST (test_tensor_filter, framework_auto_ext_tflite_model_not_found_n)
+{
+  gchar *str_launch_line;
+  const gchar *fw_name = NULL;
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH");
+  if (root_path == NULL)
+    root_path = "..";
+  gchar *test_model = g_build_filename (root_path, "tests", "test_models", "models", "mirage.tflite", NULL);
+
+  str_launch_line = g_strdup_printf ("videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter framework=auto model=%s ! tensor_sink", test_model);
+  TEST_TENSOR_FILTER_AUTO_OPTION_N (str_launch_line, fw_name)
+
+  g_free (test_model);
+}
+
+/**
+ * @brief Test framework auto detecion option in tensor-filter.
+ * @details Negative case with not supported extension
+ */
+TEST (test_tensor_filter, framework_auto_ext_tflite_not_supported_ext_n)
+{
+  const gchar *fw_name = NULL;
```

Comment:
Just asking since I am not familiar with glib yet, is it okay to not free this bit?

Reply:
free is unnecessary because there is no memory allocation